omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/UserController2_0Test.java

```diff
@@ -274,13 +285,111 @@ public void updateUser_shouldUpdateTheUserPassword() throws Exception {
+		
+		MockHttpServletResponse response = handle(newPostRequest(getPasswordURI(), "{\"password\":\""
+		        + newPassword + "\"" + "," + "\"oldPassword\":\"" + oldPassword + "\"}"));
+		assertEquals(201, response.getStatus());
+	}
+	
+	@Test
+	public void testChangeUsersOwnPasswordWithOutAuthentication() throws Exception {
+		Context.logout();
+		String oldPassword = "SomeOtherPassword123";
+		String newPassword = "newPassword9";
+		
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("You are not authenticated to change password");
+		
+		handle(newPostRequest(getPasswordURI(), "{\"password\":\""
+				+ newPassword + "\"" + "," + "\"oldPassword\":\"" + oldPassword + "\"}"));
+	}
+	@Test
+	public void testChangeUsersOwnPasswordWithIncorrectOldPassword() throws Exception {
+		setUpUser("butch");
+		
+		String wrongOldPassword = "WrongPassword";
+		String newPassword = "newPassword9";
+		
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("old.password.not.correct");
+		
+		handle(newPostRequest(getPasswordURI(), "{\"password\":\"" + newPassword + "\"" + ","
+		        + "\"oldPassword\":\"" + wrongOldPassword + "\"}"));
+		
+	}
+	
+	@Test
+	public void testUserChangeOtherUsersPassword() throws Exception {
+		User authenticatedUser = setUpUser("daemon");
+		
+		Role role = new Role("Privileged Role");
+		role.addPrivilege(new Privilege(PrivilegeConstants.EDIT_USER_PASSWORDS));
+		authenticatedUser.addRole(role);
+		
+		String newPassword = "newPassword9";
+		
+		MockHttpServletResponse response = handle(newPostRequest(getPasswordURI() + "/" + getUuid(), "{\"password\":\""
+		        + newPassword + "\"}"));
+		
+		assertEquals(200, response.getStatus());
+	}
+	
+	@Test
+	public void testUserChangeOtherUsersPasswordWithOutPrivilege() throws Exception {
+		setUpUser("daemon");
+		
+		String newPassword = "newPassword9";
+		
+		expectedException.expect(APIAuthenticationException.class);
+		expectedException.expectMessage("You doesn't have privilege to change others password");
+		
+		handle(newPostRequest(getPasswordURI() + "/" + getUuid(), "{\"password\":\"" + newPassword + "\"}"));
+		
+	}
+	
+	@Test
+	public void testThrowExceptionIfUserIsNotAvailable() throws Exception {
+		setUpUser("daemon");
+		
+		String wrongOldPassword = "WrongPassword";
+		String newPassword = "newPassword9";
+		
+		expectedException.expect(APIAuthenticationException.class);
+		expectedException.expectMessage("User not available");
+		
+		handle(newPostRequest(getPasswordURI() + "/" + "someRandomUser", "{\"password\":\"" + newPassword + "\"" + ","
+		        + "\"oldPassword\":\"" + wrongOldPassword + "\"}"));
+		
+	}
+	
+	private User setUpUser(String userName) throws Exception {
+		User user = service.getUserByUsername(userName);
+		final String newPassword = "SomeOtherPassword123";
+		
+		handle(newPostRequest(getPasswordURI() + "/" + user.getUuid(), "{\"password\":\"" + newPassword + "\"}"));
```

Comment:
FYI I would expect us to change the password with a normal call to the Java API here. (minor comment)